### PR TITLE
Updated README.md with CLI examples and additional usage instructions

### PR DESCRIPTION
**Pull request by Singhal1808**
Added Command Line Interface (CLI) examples for SFT, DPO, and Chat features. Improved the "How to Use" section by providing code examples for SFTTrainer and RewardTrainer. Included installation instructions for both Python Package and source-based installation. Refined highlights to better showcase efficiency and scalability features. Updated the repository clone instructions for working with examples. Added new links to CLI documentation and contribution guide for better navigation.

# What does this PR do?

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,34 +20,32 @@
...
-The library is built on top of [🤗 Transformers](https://github.com/huggingface/transformers) and is compatible with any model architecture available there.
+- **Efficient and scalable**: 
+    - Leverages [🤗 Accelerate](https://github.com/huggingface/accelerate) to scale from single GPU to multi-node clusters using methods like DDP and DeepSpeed.
+    - Full integration with [`PEFT`](https://github.com/huggingface/peft) enables training on large models with modest hardware via quantization and LoRA/QLoRA.
+    - Integrates [Unsloth](https://github.com/unslothai/unsloth) for accelerating training using optimized kernels.
 
+- **Command Line Interface (CLI)**: A simple interface lets you fine-tune and interact with models without needing to write code.
 
-## Highlights
+- **Custom Trainers**: Various fine-tuning methods are easily accessible via trainers like [`SFTTrainer`](https://huggingface.co/docs/trl/sft_trainer), [`DPOTrainer`](https://huggingface.co/docs/trl/dpo_trainer), and more.
```
Comment (qgallouedec) [inline README.md:36]: Is "custom" appropriate here?

File: README.md [hunk 2]
```diff
@@ -20,34 +20,32 @@
...
-### Python package
+### Python Package
 
-Install the library with `pip`:
+Install the library using `pip`:
 
 ```bash
 pip install trl
-```
+
```
Comment (qgallouedec) [inline README.md:48]: revert this

Comment (qgallouedec) [inline README.md:48]: You need to revert this change

**Comment by qgallouedec** [comment]
Can you also update the badges with the following:

```html
<p align="center">
    <a href="https://github.com/huggingface/trl/blob/main/LICENSE"><img alt="License" src="https://img.shields.io/github/license/huggingface/trl.svg?color=blue"></a>
    <a href="https://huggingface.co/docs/trl/index"><img alt="Documentation" src="https://img.shields.io/website?url=https%3A%2F%2Fhuggingface.co%2Fdocs%2Ftrl%2Findex&down_color=red&down_message=offline&up_color=blue&up_message=online"></a>
    <a href="https://github.com/huggingface/trl/releases"><img alt="GitHub release" src="https://img.shields.io/github/release/huggingface/trl.svg"></a>
</p>
```

It will remove these:

Screenshot 2024-10-08 at 22 35 28

**Review by edbeeching — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (kashif) [inline README.md:48]:
```suggestion
```
```

File: README.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <div style="text-align: center">
-<img src="https://huggingface.co/datasets/trl-internal-testing/example-images/resolve/main/images/trl_banner_dark.png">
+<img src="https://huggingface.co/datasets/trl-internal-testing/example-images/resolve/main/images/trl_banner_dark.png" alt="TRL Banner">
 </div>
 
 # TRL - Transformer Reinforcement Learning
 
 <h3 align="center">
-    <p>Full stack library to post-train large language models.</p>
+    <p>A comprehensive library to post-train large language models (LLMs).</p>
```
Comment (lewtun) [inline README.md:8]: nit: since we also support training VLMs, it's probably time to use this instead :)

```suggestion
    <p>A comprehensive library to post-train foundation models</p>
```

File: README.md [hunk 4]
```diff
@@ -20,34 +20,31 @@
     </a>
 </p>
 
+## Overview
 
-## What is it?
+TRL is a cutting-edge library designed for post-training large language models (LLMs) and diffusion models using advanced techniques like Supervised Fine-Tuning (SFT), Proximal Policy Optimization (PPO), and Direct Preference Optimization (DPO). Built on top of the [🤗 Transformers](https://github.com/huggingface/transformers) ecosystem, TRL supports a variety of model architectures and ensures seamless scalability across various hardware setups.
```
Comment (lewtun) [inline README.md:25]:
```suggestion
TRL is a cutting-edge library designed for post-training foundation models using advanced techniques like Supervised Fine-Tuning (SFT), Proximal Policy Optimization (PPO), and Direct Preference Optimization (DPO). Built on top of the [🤗 Transformers](https://github.com/huggingface/transformers) ecosystem, TRL supports a variety of model architectures and modalities, and can be scaled-up across various hardware setups.
```

**Review by lewtun — APPROVED** [review]
Thanks for improving the README @Singhal1808 ! Overall it looks good with some nits from my side and the changes that need reverting from @qgallouedec

File: README.md [hunk 5]
```diff
@@ -20,34 +20,31 @@
...
 ## Installation
 
-### Python package
+### Python Package
 
-Install the library with `pip`:
+Install the library using `pip`:
 
 ```bash
 pip install trl
```
Comment (qgallouedec) [inline README.md:47]:
````suggestion
pip install trl
```
````
